website/content/docs/security/acl/acl-roles.mdx

```diff
@@ -0,0 +1,378 @@
+---
+layout: docs
+page_title: Roles
+description: >-
+  This topic describes roles within the access control list (ACL) system. A role is a named set of policies and service identities.
+  They are intended to
+---
+
+# Roles
+
+A role is a collection of policies that your ACL administrator can link to a token.
+They enable you to reuse policies by decoupling the policies from the token distributed to team members.
+Instead, the token is linked to the role, which is able to hold several policies that can be updated asynchronously without distributing new tokens to users.
+As a result, roles can provide a more convenient authentication infrastrcture than creating unique policies and tokens for each requester.
+
+## Workflow Overview
+
+Roles are configurations link several policies to a token. The following procedure describes the workflow for implementing roles.
+
+1. Assemble rules into policies (see [Policies](/docs/security/acl/acl-policies)) and register them in Consul.
+1. Define a role and include the policy IDs or names.
+1. Register the role in Consule and link it to a token.
+1. Distribute the tokens to users for implementation.
+
+## Creating Roles
+
+Creating roles is commonly the responsibility of the Consul ACLs administrator.
+Roles have several attributes, including service identities and node identities.
+Refer to the following documentation for details:
+
+- [Role Attributes](#role-attributes)
+- [Service Identities](#service-identities)
+- [Node Identities](#node-identities)
+
+Use the Consul command line or API endpoint to create roles.
+
+### Command Line
+
+Issue the `consul acl role create` command to create roles. In the following example, a role named `crawler` is created that contains a policy named `crawler-kv` and a policy named `crawler-key`.
+
+```shell-session
+$ consul acl role create -name "crawler" -description "web crawler role" -policy-name "crawler-kv" -policy-name "crawler-key"
+```
+
+Refer to the [command line documentation](/command/acl/role) for details.
+
+### API
+
+Make a `PUT` call to the `acl/role` endpoint and specify the role configuration in the payload to create roles. You can save the role definition in a JSON file or use escaped JSON in the call. In the following example call, the payload is defined externally.
+
+```shell-session
+$ curl -X PUT --data @payload.json http://127.0.0.1:8500/v1/acl/role
+```
+
+Refer to the [API documentation](/api-docs/acl/roles) for details.
+
+## Role Attributes
```

Comment:
similar opinion about this not being the right place to document role attributes.